### DISCUSSION TITLE
How can I make an array in the database?

### DISCUSSION THREAD

**Discussion by nailuj29** [supabase/supabase#1004] [Questions]
I am porting a firebase project I made over to supabase, but it uses arrays in the database, and it doesn't look like this is possible in supabase. Am I missing something?

**Comment by joshnuss** [comment]
Hi @nailuj29gaming,

Here's how you can use arrays:

When defining a table, specify that a column should be an array by adding square brackets after the data type `[]`:

```sql
create table vehicles (
  id serial primary key,
  name text,
  -- the square brackets define an array:
  features text[]
)
```

Alternatively, if you're creating via the UI, click the "Define as array" checkbox when adding a column.

Then to insert an array into the table:

```sql
insert into vehicles (name, features) values ('tesla', '{"electric", "fast"}')
```

and it works with the client API too:

```javascript
supabase.from('vehicles')
  .insert({name: "tesla", features: ["electric", "fast"]})
```

**Reply by nailuj29** [reply]
Sorry, I didn't phrase my question well. How can I make an array of objects with multiple values? I assume I can use foreign keys, but I don't have that much experience with PostgreSQL

**Reply by burggraf** [reply]
Unless I'm missing something here, I think what you want is a JSONB field.  You can throw JSON into it, and it'll even be indexable and searchable!
